### PR TITLE
Dev/Staging k8s 1.31.0 20241024 and VPN Admin script

### PR DESCRIPTION
**Pull request by ben851**
# Summary | Résumé

This will update Staging K8s to latest.

I've also created a script that will automatically update (or create) VPN configs so that we don't have to continually download them when dev gets destroyed/created.

## Related Issues | Cartes liées

Chore
 
## Before merging this PR

Read code suggestions left by the
[cds-ai-codereviewer](https://github.com/cds-snc/cds-ai-codereviewer/) bot. Address
valid suggestions and shortly write down reasons to not address others. To help
with the classification of the comments, please use these reactions on each of the
comments made by the AI review:

| Classification      | Reaction | Emoticon |
|---------------------|----------|----------|
| Useful              | +1       | 👍        |
| Noisy               | eyes     | 👀        |
| Hallucination       | confused | 😕        |
| Wrong but teachable | rocket   | 🚀        |
| Wrong and incorrect | -1       | 👎        |

The classifications will be extracted and summarized into an analysis of how helpful
or not the AI code review really is.

## Test instructions | Instructions pour tester la modification

Tested in dev
TF Apply works in staging
Smoke test staging

## Release Instructions | Instructions pour le déploiement

None.

## Reviewer checklist | Liste de vérification du réviseur

* [ ] This PR does not break existing functionality.
* [ ] This PR does not violate GCNotify's privacy policies.
* [ ] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
* [ ] This PR does not significantly alter performance.
* [ ] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: scripts/getVPNConfig.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+done
+shift "$(($OPTIND -1))"
+
+echo $FILE_NAME
+#Find the logged in user
+LOGGED_USER=$(stat -f %Su /dev/console)
+export FULL_PATH="/Users/$LOGGED_USER/.config/AWSVPNClient/OpenVpnConfigs/$FILE_NAME"
+
+echo "Getting SAML VPN Config for AWS Profile: $AWS_PROFILE"
+VPN_ID=$(aws ec2 describe-client-vpn-endpoints --query 'ClientVpnEndpoints[? Description == `private-subnets`].ClientVpnEndpointId' --output text)
```
[the remark this answers is not in the document]
Comment (ben851) [reply scripts/getVPNConfig.sh:31]: This is set by AWS CLI and will always be present.

File: scripts/getVPNConfig.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+aws ec2 export-client-vpn-client-configuration --client-vpn-endpoint-id $VPN_ID --output text | sed '/auth-federate/d' > /Users/$LOGGED_USER/.config/AWSVPNClient/OpenVpnConfigs/"$FILE_NAME"
```
[the remark this answers is not in the document]
Comment (ben851) [reply scripts/getVPNConfig.sh:33]: I think this is ok as is.

File: scripts/getVPNConfig.sh [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+cp /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles.bak
+
+if [[ "$CREATE" == "true" ]]; then
+  echo "Creating new VPN Config"
+  mkdir -p /Users/$LOGGED_USER/.config/AWSVPNClient/OpenVpnConfigs
+
+  jq -c '.ConnectionProfiles[.ConnectionProfiles | length] |= . + {
+        "ProfileName": env.FILE_NAME, 
+        "OvpnConfigFilePath": "/Users/env.LOGGED_USER/.config/AWSVPNClient/OpenVpnConfigs/env.FILE_NAME",
```
[the remark this answers is not in the document]
Comment (ben851) [reply scripts/getVPNConfig.sh:43]: This is static and works as expected - don't want to touch it.

File: scripts/getVPNConfig.sh [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        "CvpnEndpointId": env.VPN_ID,
+        "CvpnEndpointRegion": "ca-central-1",
+        "CompatibilityVersion": "2",
+        "FederatedAuthType": 1
+  }' /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles > /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles.tmp
```
[the remark this answers is not in the document]
Comment (ben851) [reply scripts/getVPNConfig.sh:48]: Inaccurate - doing two JQ's

File: scripts/getVPNConfig.sh [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        "OvpnConfigFilePath": "/Users/env.LOGGED_USER/.config/AWSVPNClient/OpenVpnConfigs/env.FILE_NAME",
+        "CvpnEndpointId": env.VPN_ID,
+        "CvpnEndpointRegion": "ca-central-1",
+        "CompatibilityVersion": "2",
+        "FederatedAuthType": 1
+  }' /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles > /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles.tmp   
+  
+
+
+  jq -c '( .ConnectionProfiles[] | select(.ProfileName == env.FILE_NAME) | .OvpnConfigFilePath ) |= env.FULL_PATH' /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles.tmp  > /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles.tmp1
```
[the remark this answers is not in the document]
Comment (ben851) [reply scripts/getVPNConfig.sh:52]: Inaccurate - doing two JQ's

File: scripts/getVPNConfig.sh [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+  rm /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles.tmp
+  mv /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles.tmp1 /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles
+
+fi
+
+if [[ "$UPDATE" == "true" ]]; then
+  echo "Updating VPN Config"
+
+  jq -c '( .ConnectionProfiles[] | select(.ProfileName == env.FILE_NAME) | .CvpnEndpointId ) |= env.VPN_ID' /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles > /Users/$LOGGED_USER/.config/AWSVPNClient/ConnectionProfiles.tmp
```
[the remark this answers is not in the document]
Comment (ben851) [reply scripts/getVPNConfig.sh:62]: This is ok as is

File: scripts/getVPNConfig.sh [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      exit 1
+      ;;
+  esac
+done
+shift "$(($OPTIND -1))"
+
+echo "$FILE_NAME"
+#Find the logged in user
+LOGGED_USER=$(stat -f %Su /dev/console)
+# Get full path
```
Comment (sastels) [inline scripts/getVPNConfig.sh:28]: These two comments seem unnecessary but 🤷

Comment (ben851) [reply scripts/getVPNConfig.sh:28]: I've never been accused of commenting too much before

**Review by sastels — APPROVED** [review]

**Pull request merged**